xbmc/filesystem/AddonsDirectory.cpp

```diff
@@ -118,6 +133,37 @@ static void SetUpdateAvailProperties(CFileItemList &items)
   }
 }
 
+
+// Creates categories for addon types, if we have any addons with that type.
+static void GenerateTypeListing(
+  const CURL& path,
+  const std::vector<TYPE>& types,
+  const VECADDONS& addons,
+  CFileItemList& items)
+{
+  for (const auto& type : types)
```

Comment:
Ah just realized that this won't work for `CPluginSource`'s overloaded implementation of `IsType()` because it needs to check against multiple possible types. So using `IsLookAndFeelType()` might not work the same as `CPluginSource::IsType()`.